PushSharp.Core/ServiceBroker.cs

```diff
@@ -179,6 +179,9 @@ public void Start ()
 
                             // Let's wait for the continuation not the task itself
                             toSend.Add (cont);
+                            
+                            // release references to completed tasks
+                            toSend.RemoveAll(task => task.IsCompleted);
```

Comment:
I got pinged from a comment I opened 4 years ago.

This should help but I believe you want to check if Task.IsFaulted (as well) GC will throw a UnobservedTaskException if the error is not handled. Or you could add an AppDomain UnobservedTaskException handler.